dhis-2/dhis-web-api-test/src/test/java/org/hisp/dhis/webapi/controller/BulkPatchSharingTest.java

```diff
@@ -0,0 +1,240 @@
+    }
+
+    @Test
+    public void testApplyPatches()
+        throws IOException
+    {
+        assertStatus( HttpStatus.CREATED, POST( "/users", createUser( "userA", userAId ) ) );
+        assertStatus( HttpStatus.CREATED, POST( "/users", createUser( "userB", userBId ) ) );
+        userCId = assertStatus( HttpStatus.CREATED, POST( "/users", createUser( "userC" ) ) );
+
+        assertStatus( HttpStatus.CREATED,
+            POST( "/dataSets", jsonMapper.writeValueAsString( createDataSet( 'A', dsIdA, userCId ) ) ) );
+        assertStatus( HttpStatus.CREATED,
+            POST( "/dataElements", jsonMapper.writeValueAsString( createDataElement( 'A', deAId, userCId ) ) ) );
+        assertStatus( HttpStatus.CREATED,
+            POST( "/dataElements", jsonMapper.writeValueAsString( createDataElement( 'B', deBId, userCId ) ) ) );
+
+        String payload = IOUtils.toString( new ClassPathResource( "patch/bulk_sharing_patches.json" ).getInputStream(),
+            StandardCharsets.UTF_8 );
+        assertStatus( HttpStatus.OK, PATCH( "/metadata/sharing", payload ) );
+
+        JsonIdentifiableObject savedDeA = GET( "/dataElements/{uid}", deAId ).content( HttpStatus.OK )
+            .as( JsonIdentifiableObject.class );
+        JsonIdentifiableObject savedDeB = GET( "/dataElements/{uid}", deBId ).content( HttpStatus.OK )
+            .as( JsonIdentifiableObject.class );
+        JsonIdentifiableObject savedDsA = GET( "/dataSets/{uid}", dsIdA ).content( HttpStatus.OK )
+            .as( JsonIdentifiableObject.class );
+
+        assertEquals( 2, savedDeA.getSharing().getUsers().size() );
+        assertEquals( 2, savedDeB.getSharing().getUsers().size() );
+        assertEquals( 2, savedDsA.getSharing().getUsers().size() );
+
+        assertNotNull( savedDeA.getSharing().getUsers().get( userAId ) );
+        assertNotNull( savedDeB.getSharing().getUsers().get( userAId ) );
+        assertNotNull( savedDsA.getSharing().getUsers().get( userAId ) );
+
+        assertNotNull( savedDeA.getSharing().getUsers().get( userBId ) );
+        assertNotNull( savedDeB.getSharing().getUsers().get( userBId ) );
+        assertNotNull( savedDsA.getSharing().getUsers().get( userBId ) );
+    }
+
+    @Test
+    public void testApplyPatchesInvalidClass()
+        throws IOException
+    {
+        String payload = IOUtils.toString(
+            new ClassPathResource( "patch/bulk_sharing_patches_invalid_class.json" ).getInputStream(),
+            StandardCharsets.UTF_8 );
+
+        HttpResponse response = PATCH( "/metadata/sharing", payload );
+        assertEquals( HttpStatus.CONFLICT, response.status() );
+        assertEquals( "Sharing is not enabled for this object `organisationUnit`", getFirstErrorMessage( response ) );
+    }
+
+    private String getFirstErrorMessage( HttpResponse response )
+    {
+        return response.error().get( "response.typeReports[0].objectReports[0].errorReports[0].message" )
```

Comment:
FYI: There is also the underlying `JsonNode` tree API which has a `find` method that could be what you want. But if you always want to get the first (and I assume only) error this is the way.

Reply:
ok noted